raddb/sites-available/coa-relay

```diff
@@ -0,0 +1,289 @@
+# -*- text -*-
+######################################################################
+#
+#  Sample site for receiving a CoA or Disconnect-Request packet for a user
+#  and/or session, looking up matching sessions in the radacct table, then
+#  relaying the packet towards the owning NAS for each session.
+#
+#  The process is as follows:
+#
+#    - A CoA/Disconnect-Request is received by FreeRADIUS.
+#    - The radacct table is searched for active sessions that match each of
+#      the provided identifier attributes: User-Name, Acct-Session-Id. The
+#      search returns the owning NAS and Acct-Unique-Id for the matching
+#      session/s.
+#    - The original CoA/Disconnect-Request content is written to a detail file
+#      with custom attributes representing the NAS and Acct-Session-Id.
+#    - A detail reader follows the file and originates CoA/Disconenct-Requests
+#      containing the original content, relayed to the corresponding NAS for
+#      each session using the custom attributes.
+#
+#  This simplifies scripting directly against a set of NAS devices since a
+#  script need only send a single CoA/Disconenct to FreeRADIUS which will
+#  then:
+#
+#    - Lookup all active sessions belonging to a user, in the case that only a
+#      User-Name attribute is provided in the request
+#    - Handle routing of the request to the correct NAS, in the case of a
+#      multi-NAS setup
+#
+#  For example, to Disconnect a specific session:
+#
+#    $ echo 'Acct-Session-Id = "769df3 312343"' | \
+#      radclient 127.0.0.1 disconnect testing123
+#
+#  To perform a CoA update of all active sessions belonging to a user:
+#
+#    $ cat <<EOF | radclient 127.0.0.1 coa testing123
+#      User-Name = bob
+#      Cisco-AVPair = "subscriber:sub-qos-policy-out=q_out_uncapped"
+#      EOF
+#
+#  In addition to configuring and activating this site...
+#
+#  Custom attributes must be created in the local dictionary:
+#
+#    ATTRIBUTE  COA-Packet-Type            3010  string
+#    ATTRIBUTE  COA-Packet-DST-IP-Address  3011  ipaddr
+#    ATTRIBUTE  COA-Packet-DST-Port        3012  integer
+#    ATTRIBUTE  COA-Acct-Session-Id        3013  string
+#
+#  A detail writer module must be configured:
+#
+#    detail detail_coa {
+#        filename = ${radacctdir}/detail_coa
+#        escape_filenames = no
+#        permissions = 0600
+#        header = "%t"
+#        locking = yes
+#    }
+#
+
+
+#  Listen on a local CoA port.
+#
+#  This uses the normal set of clients, with the same secret as for
+#  authentication and accounting.
+#
+listen {
+	type = coa
+	ipaddr = 127.0.0.1
+	port = 3799
+	virtual_server = coa
+}
+
+#
+#  Receive CoA/Disconnect, lookup sessions, write them to a detail file
+#
+server coa {
+	#  When a packet is received, it is processed through the
+	#  recv-coa section.  This applies to *both* CoA-Request and
+	#  Disconnect-Request packets.
+	recv-coa {
+
+		# Record the original request type (CoA or Disconnect)
+		#
+		switch Packet-Type {
```

Comment:
This could just be

```
update request {
  CoA-Packet-Type := "%{Packet-Type}"
}
```

Reply:
Thanks. I wrote this out noddy fashion after initially struggling with syntax.